lib/broadway_kafka/producer.ex

```diff
@@ -524,30 +529,70 @@ defmodule BroadwayKafka.Producer do
   end
 
   defp maybe_schedule_poll(state, interval) do
-    %{buffer: buffer, demand: demand, acks: acks, receive_timer: receive_timer} = state
+    %{
+      buffers: buffers,
+      demand: demand,
+      max_demand: max_demand,
+      acks: acks,
+      receive_timer: receive_timer
+    } = state
 
-    case dequeue_many(buffer, acks, demand, []) do
-      {acks, 0, events, buffer} ->
-        {:noreply, events, %{state | demand: 0, buffer: buffer, acks: acks}}
+    {key, next_queue, next_index} = find_next_queue(buffers, state.buffer_index)
 
-      {acks, demand, events, buffer} ->
-        receive_timer = receive_timer || schedule_poll(state, interval)
+    num_to_dequeue = min(demand, max_demand)
```

Comment:
What is the correct approch? maybe the incoming_demand is only 5 but since we don't pass it along. Here we are going to send max_demand. 

e.g: current demand is 20 and max_demand is 10.
this function is called by because handle_demand was called with 5 as incoming_demand.
Sending 10 events causes that we overload the PartitionDipatacher buffer